src/main/scala/io/iohk/ethereum/jsonrpc/server/controllers/JsonRpcControllerCommon.scala

```diff
@@ -0,0 +1,130 @@
+package io.iohk.ethereum.jsonrpc.server.controllers
+
+import java.util.concurrent.TimeUnit
+
+import com.typesafe.config.{Config => TypesafeConfig}
+import io.iohk.ethereum.jsonrpc.JsonRpcError.{InternalError, MethodNotFound}
+import io.iohk.ethereum.jsonrpc.{JsonRpcControllerMetrics, JsonRpcError, JsonRpcRequest, JsonRpcResponse}
+import io.iohk.ethereum.jsonrpc.serialization.{JsonEncoder, JsonMethodDecoder}
+import io.iohk.ethereum.jsonrpc.server.http.JsonRpcHttpServer.JsonRpcHttpServerConfig
+import io.iohk.ethereum.jsonrpc.server.ipc.JsonRpcIpcServer.JsonRpcIpcServerConfig
+import io.iohk.ethereum.utils.Logger
+import org.json4s.{DefaultFormats, native}
+import org.json4s.JsonDSL._
+import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.duration.FiniteDuration
+import scala.util.{Failure, Success}
+
+trait JsonRpcControllerCommon extends Logger {
```

Comment:
Maybe `JsonRpcBaseController` sounds better? WDYT?

Reply:
And I think it could be self-trait with `ApisBuilder`

Reply:
Done